compiler/packages/eslint-plugin-react-compiler/src/rules/ReactCompilerRule.ts

```diff
@@ -161,9 +160,16 @@ const rule: Rule.RuleModule = {
               detail.loc != null && typeof detail.loc !== "symbol"
                 ? ` (@:${detail.loc.start.line}:${detail.loc.start.column})`
                 : "";
+            const firstLineLoc = {
+              start: event.fnLoc.start,
+              end: {
+                line: event.fnLoc.start.line,
+                column: 10e3,
+              },
+            };
```

Comment:
Slightly hacky, but the alternative is to attach another loc object to events (e.g. function identifier, parameters, etc). This felt cleaner, as not all functions we compile have id or param nodes.

Reply:
Worth noting that in Flow we have a `sig_loc` on functions, which is the location of, effectively, everything in a function other than its body (`function` keyword, name, type parameters, parameters including parentheses, => for arrow functions, return type annotation). This has been a useful thing for us to have in a lot of error messages and the compiler might benefit from something similar.

Reply:
Ohhh i've wanted that a few times